### PR TITLE
[BM-622] Handle sdp set error on baby device

### PR DESCRIPTION
**Pull request by gabrysgab**
### Task

 - [JIRA](https://netguru.atlassian.net/browse/BM-622)
 
### Description

When baby device encounters error on setting sdp description
we start the machine learning audio recording just like in end of the stream
and send a message to the parent device about the error. Parent device
shows error snackbar and closes the stream.

### CODE REVIEW

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt [hunk 1]
```diff
@@ -158,7 +158,14 @@ class WebRtcManager constructor(
...
                 .doOnComplete { Timber.d("Answer set as a local description.") }
-                .subscribe()
+                .subscribeBy(onError = {
+                    connectionObserver.onSetDescriptionError()
+                    sendMessage(
+                        Message(
+                            sdpError = it.message
+                        )
+                    )
+                    Timber.e(it)})
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt#L168"></span>Missing spacing before “}”</td>
    </tr>
  </tbody>
</table>

**Comment by netguru-hound** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          4 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L22">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L22</a> - RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L25">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L25</a> - RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L46">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L46</a> - RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L49">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/PeerConnectionExtensions.kt#L49</a> - RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/1a0452485cf5a8f89a82a51c8b21e5120d8c1d8a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt#L46">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager.kt#L46</a> - This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:link: <a href="https://netguru.atlassian.net/browse/BM-622">BM-622</a></td>
    </tr>
  </tbody>
</table>

### JaCoCO Code Coverage 10.27% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `co/netguru/baby/monitor/client/feature/communication/webrtc/server/WebRtcManager` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/websocket/Message` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragmentViewModel` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClient` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/client/RtcClientMessageController` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/observers/ConnectionObserver` | 0% | 0% | :white_check_mark: |


#### APK file size

| size |
| --- |
| 29.9MB |



#### Number of method references

| Dex file | count |
| --- | --- |
| classes.dex | 39362 |



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment.kt [hunk 2]
```diff
@@ -98,7 +100,14 @@ class ClientLiveCameraFragment : BaseDaggerFragment() {
...
                 View.VISIBLE
+            RtcConnectionState.Error -> handleBabyDeviceSdpError()
             else -> Unit
         }
     }
+
+    private fun handleBabyDeviceSdpError() {
+        showSnackbarMessage(R.string.stream_error)
+        requireActivity()
+            .findNavController(R.id.clientDashboardNavigationHostFragment).navigateUp()
```
Comment (Tarapaty) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment.kt]: Would regular `requireActivity().onBackPressed()` not be enough in this use case?

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment.kt [hunk 3]
```diff
@@ -5,8 +5,10 @@ import android.view.View
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.findNavController
```
Comment (netguru-hound) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/23ee4f2e486640b6252e077036ee5f11a86fa33a/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/livecamera/ClientLiveCameraFragment.kt#L8"></span>Unused import</td>
    </tr>
  </tbody>
</table>

**Pull request merged**